Feature unsealed_read_write should work without std

Run `cargo build --no-default-features --features unsealed_read_write`.

It fails with:

```
error[E0599]: no variant or associated item named `Io` found for type `error::ErrorCode` in the current scope
   --> src/error.rs:53:30
    |
53  |             code: ErrorCode::Io,
    |                              ^^ variant or associated item not found in `error::ErrorCode`
...
263 | pub(crate) enum ErrorCode {
    | ------------------------- variant or associated item `Io` not found here

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
error: Could not compile `serde_cbor`.

To learn more, run the command again with --verbose.
```

@baloo now we know why `Error::Io` is needed.

--- a/.travis.yml
+++ b/.travis.yml
@@ -18,3 +18,4 @@ script:
   - cargo build --no-default-features --target thumbv7em-none-eabihf # Test we can build a platform that does not have std.
   - cargo test --no-default-features --lib --tests # Run no_std tests
   - cargo build --features unsealed_read_write # The crate should still build when the unsealed_read_write feature is enabled.
+  - cargo build --no-default-features --features unsealed_read_write # The crate should still build when the unsealed_read_write feature is enabled and std disabled.

--- a/src/error.rs
+++ b/src/error.rs
@@ -130,6 +130,8 @@ impl Error {
             ErrorCode::Message => Category::Data,
             #[cfg(feature = "std")]
             ErrorCode::Io(_) => Category::Io,
+            #[cfg(not(feature = "std"))]
+            ErrorCode::Io => Category::Io,
             ErrorCode::ScratchTooSmall => Category::Io,
             ErrorCode::EofWhileParsingValue
             | ErrorCode::EofWhileParsingArray
@@ -267,6 +269,9 @@ pub(crate) enum ErrorCode {
     Message,
     #[cfg(feature = "std")]
     Io(io::Error),
+    #[allow(unused)]
+    #[cfg(not(feature = "std"))]
+    Io,
     ScratchTooSmall,
     EofWhileParsingValue,
     EofWhileParsingArray,
@@ -292,6 +297,8 @@ impl fmt::Display for ErrorCode {
             ErrorCode::Message => f.write_str("Unknown error"),
             #[cfg(feature = "std")]
             ErrorCode::Io(ref err) => fmt::Display::fmt(err, f),
+            #[cfg(not(feature = "std"))]
+            ErrorCode::Io => f.write_str("Unknown I/O error"),
             ErrorCode::ScratchTooSmall => f.write_str("Scratch buffer too small"),
             ErrorCode::EofWhileParsingValue => f.write_str("EOF while parsing a value"),
             ErrorCode::EofWhileParsingArray => f.write_str("EOF while parsing an array"),

